### PR TITLE
util/collate: migrate to testify

### LINKED ISSUE #26172: migrate test-infra to testify for `util/collate` pkg

### PR DESCRIPTION
**Pull request by ofey404**
### What problem does this PR solve?

Issue Number: close #26172  

### What is changed and how it works?

What's Changed: Removed `pingcap/check` for utils/collate, moved to testify.

P.S. I removed `testleak` below in test functions, for leak test is done in `goleak.VerifyTestMain`. I think it's okay.

```golang
defer testleak.AfterTest(c)()
```

### Release note 

- No release note.

### CODE REVIEW

**Comment by ofey404** [comment]
P.P.S: I use `reflect.DeepEqual` directly to replace `c.Assert(.., DeepEqual, ..)`.

File: util/collate/collate_test.go [hunk 1]
```diff
@@ -14,23 +14,13 @@
 package collate
 
 import (
+	"reflect"
```
Comment (tisonkun) [inline util/collate/collate_test.go:17]:
```suggestion
	"fmt"
```

File: util/collate/collate_test.go [hunk 2]
```diff
@@ -14,23 +14,13 @@
...
 	"testing"
 
 	. "github.com/pingcap/check"
```
Comment (tisonkun) [inline util/collate/collate_test.go:20]:
```suggestion
```

File: util/collate/collate_test.go [hunk 3]
```diff
@@ -42,22 +32,21 @@ type keyTable struct {
 	Expect []byte
 }
 
-func testCompareTable(table []compareTable, collate string, c *C) {
-	for i, t := range table {
-		comment := Commentf("%d %v %v", i, t.Left, t.Right)
-		c.Assert(GetCollator(collate).Compare(t.Left, t.Right), Equals, t.Expect, comment)
+func testCompareTable(tables []compareTable, collate string, t *testing.T) {
+	for i, table := range tables {
+		comment := Commentf("%d %v %v", i, table.Left, table.Right)
```
Comment (tisonkun) [inline util/collate/collate_test.go:37]:
```suggestion
		comment := fmt.Sprintf("%d %v %v", i, table.Left, table.Right)
```

File: util/collate/collate_test.go [hunk 4]
```diff
@@ -42,22 +32,21 @@ type keyTable struct {
...
 	}
 }
 
-func testKeyTable(table []keyTable, collate string, c *C) {
-	for i, t := range table {
-		comment := Commentf("%d %s", i, t.Str)
-		c.Assert(GetCollator(collate).Key(t.Str), DeepEquals, t.Expect, comment)
+func testKeyTable(tables []keyTable, collate string, t *testing.T) {
+	for i, table := range tables {
+		comment := Commentf("%d %s", i, table.Str)
```
Comment (tisonkun) [inline util/collate/collate_test.go:44]:
```suggestion
		comment := fmt.Sprintf("%d %s", i, table.Str)
```

File: util/collate/collate_test.go [hunk 5]
```diff
@@ -42,22 +32,21 @@ type keyTable struct {
...
+		require.True(t, reflect.DeepEqual(GetCollator(collate).Key(table.Str), table.Expect), comment)
```
Comment (tisonkun) [inline util/collate/collate_test.go:45]:
```suggestion
		require.Equal(t, table.Expect, GetCollator(collate).Key(table.Str), comment)
```

File: util/collate/collate_test.go (hunk 1 above)
Comment (tisonkun) [reply util/collate/collate_test.go:17]: replace `Commentf` from pingcap/check

File: util/collate/collate_test.go (hunk 2 above)
Comment (tisonkun) [reply util/collate/collate_test.go:20]: our target is to remove pingcap/check dependency

File: util/collate/collate_test.go (hunk 5 above)
Comment (tisonkun) [reply util/collate/collate_test.go:45]: `require.Equal` works. You can take a look at their implementation.

**Review by tisonkun — CHANGES_REQUESTED** [review]
Thanks for your contribution @ofey404 !

We are going to remove "pingcap/check" so we don't use `Commentf` from there. Also other comments inline.

**Comment by tisonkun** [comment]
> P.S. I removed testleak below in test functions, for leak test is done in goleak.VerifyTestMain. I think it's okay.

You're right.

File: util/collate/collate_test.go (hunk 5 above)
Comment (ofey404) [reply util/collate/collate_test.go]: Yes, they use `reflect.DeepEqual` in `ObjectsAreEqual()` function.

File: util/collate/collate_test.go (hunk 2 above)
Comment (ofey404) [reply util/collate/collate_test.go]: Done.

**Comment by ofey404** [comment]
It seems that it's a timeout in `testPessimisticSuite`.

```
[2021-07-26T09:13:09.247Z] PASS: pessimistic_test.go:2335: testPessimisticSuite.TestAmendWithColumnTypeChange	5.967s

[2021-07-26T09:13:12.650Z] PASS: pessimistic_test.go:2568: testPessimisticSuite.TestAsyncCommitCalTSFail	1.288s

[2021-07-26T09:13:22.632Z] panic: test timed out after 10m0s
```

Should I retrigger the test? Or there is something I can dig in further?

**Comment by tisonkun** [comment]
@ofey404 I think it is a global timeout under `FAIL	github.com/pingcap/tidb/session	600.083s`. We are tracking these kinds of issues in #25899 #26575 . You may take a look.

For this specific PR, you can see the test passed now. Hopefully we reach a stable status of tests.

**Review by tisonkun — APPROVED** [review]
LGTM

**Comment by xhebox** [comment]
@tisonkun jiwei will be back on august, if I did not remember it wrong.

**Comment by tisonkun** [comment]
> will be back on august

Got it.

/uncc @xiongjiwei

**Review by xhebox — APPROVED** [review]

File: util/collate/collate_test.go [hunk 6]
```diff
@@ -162,71 +147,70 @@ func (s *testCollateSuite) TestUnicodeCICollator(c *C) {
...
 	defer SetNewCollationEnabledForTest(false)
 
 	SetNewCollationEnabledForTest(true)
-	c.Assert(NewCollationEnabled(), Equals, true)
+	require.True(t, NewCollationEnabled())
 }
 
-func (s *testCollateSuite) TestRewriteAndRestoreCollationID(c *C) {
+func TestRewriteAndRestoreCollationID(t *testing.T) {
 	SetNewCollationEnabledForTest(true)
```
Comment (mmyj) [inline util/collate/collate_test.go:162]: Should we need to use `defer SetNewCollationEnabledForTest(false)` here?

Comment (xhebox) [reply util/collate/collate_test.go:162]: There is `SetNewCollationEnabledForTest(false)` later.

Comment (mmyj) [reply util/collate/collate_test.go:162]: So L177 is useless?

Comment (xhebox) [reply util/collate/collate_test.go:162]: I think it still makes sense before `SetNewCollationEnabledForTest(false)`. IMO, they are two groups of tests with new collation enabled/disabled.

Comment (mmyj) [reply util/collate/collate_test.go:162]: I think reset it to false seems better when the asserts L163-L166 are failed.

**Review by mmyj — APPROVED** [review]

**Pull request merged**